File: project/.sdk/tm/rb/utility/prepare_auth.rb
# ProjectName SDK utility: prepare_auth
require_relative 'struct/voxgig_struct'
module ProjectNameUtilities
  HEADER_AUTH = "authorization"
  OPTION_APIKEY = "apikey"
  NOT_FOUND = "__NOTFOUND__"

  PrepareAuth = ->(ctx) {
    spec = ctx.spec
    return nil, ctx.make_error("auth_no_spec", "Expected context spec property to be defined.") unless spec

    headers = spec.headers
    options = ctx.client.options_map
    apikey = VoxgigStruct.getprop(options, OPTION_APIKEY, NOT_FOUND)

    if apikey.is_a?(String) && apikey == NOT_FOUND
      headers.delete(HEADER_AUTH)
    else
      auth_prefix = VoxgigStruct.getpath(options, "auth.prefix") || ""
      apikey_val = apikey.is_a?(String) ? apikey : ""
      headers[HEADER_AUTH] = "#{auth_prefix} #{apikey_val}"
    end

    return spec, nil
  }
end

File: project/.sdk/tm/py/utility/prepare_auth.py
# ProjectName SDK utility: prepare_auth

from __future__ import annotations
from utility.voxgig_struct import voxgig_struct as vs

HEADER_AUTH = "authorization"
OPTION_APIKEY = "apikey"
NOT_FOUND = "__NOTFOUND__"


def prepare_auth_util(ctx):
    spec = ctx.spec
    if spec is None:
        return None, ctx.make_error("auth_no_spec",
            "Expected context spec property to be defined.")

    headers = spec.headers
    options = ctx.client.options_map()

    apikey = vs.getprop(options, OPTION_APIKEY, NOT_FOUND)

    if isinstance(apikey, str) and apikey == NOT_FOUND:
        headers.pop(HEADER_AUTH, None)
    else:
        auth_prefix = ""
        ap = vs.getpath(options, "auth.prefix")
        if isinstance(ap, str):
            auth_prefix = ap
        apikey_val = ""
        if isinstance(apikey, str):
            apikey_val = apikey
        headers[HEADER_AUTH] = auth_prefix + " " + apikey_val

    return spec, None
